vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -675,15 +675,24 @@ func (s *GenericAPIServer) Eventf(eventType, reason, messageFmt string, args ...
 	}
 }
 
-// terminationLoggingListener wraps the given listener and logs new connections
-// after the stopCh has been closed, i.e. when termination has begun.
+// terminationLoggingListener wraps the given listener to mark late connections
+// as such, identified by the remote address. In parallel, we have a filter that
+// logs bad requests through these connections. We need this filter to get
+// access to the http path in order to filter out healthz or readyz probes that
+// are allowed at any point during termination.
+//
+// Connections are late after the lateStopCh has been closed.
 type terminationLoggingListener struct {
```

Comment:
Could you move `terminationLoggingListener` and `WithLateConnectionFilter` to a new file? We could avoid potential conflicts on rebasing.